Assets/Scripts/PotionCollection.cs

```diff
@@ -0,0 +1,238 @@
+                {
+                    Color loadedCol;
+
+                    GameObject loadedPotionObject = Instantiate(_potionObject);
+
+                    if (ColorUtility.TryParseHtmlString(potionSave.Color[i], out loadedCol))
+                    {
+                        loadedPotionObject.GetComponent<Image>().color = loadedCol;
+                    }
+
+                    Potion loadedPotion = loadedPotionObject.GetComponent<Potion>();
+                    loadedPotion.transform.position = _potionSlots[i].transform.position;
+                    loadedPotion.transform.SetParent(_canvas.transform, true);
+                    _potionSlots[i].SetPotion(loadedPotion);
+
+                    loadedPotion.Color = loadedCol;
+                    loadedPotion.SetCount(potionSave.Count[i]);
+                    loadedPotion.PotionSlot = _potionSlots[i];
+
+                    loadedPotion.AddEvents(_disableEvent, _enableEvent, _decreaseEvent);
+
+                    _varietyAmount++;
+                }
+            }
+
+            Debug.Log(contents);
+        }
+    }
+
+    public void Generate()
+    {
+        if(DateTime.Now < _nextDateTime)
+        {
+            Debug.Log("Please wait a little longer...");
+            return;
+        }
+
+        if (_varietyAmount < _potionSlots.Count)
+        {
+            Color newColor = Color.HSVToRGB(UnityEngine.Random.Range(0f, 1f), 1, 1);
+
+            if (!CheckIfColorUsed(newColor))
+            {
+                SpawnNewPotion(newColor);
+            }
+        }
+
+        else
+        {
+            int index = UnityEngine.Random.Range(0, _potionSlots.Count);
+            _potionSlots[index].GetPotion().IncreaseCount();
+        }
+
+        _nextDateTime = DateTime.Now.AddSeconds(_waitTime);
+    }
+
+    private void SpawnNewPotion(Color newColor)
+    {
+        GameObject newPotionObject = Instantiate(_potionObject);
+        int index = CheckAvailableSlot();
+
+        newPotionObject.GetComponent<Image>().color = newColor;
+
+        Potion newPotion = newPotionObject.GetComponent<Potion>();
+        newPotion.PotionSlot = _potionSlots[index];
+        newPotion.Color = newColor;
+        newPotion.transform.position = _potionSlots[index].transform.position;
+        newPotion.transform.SetParent(_canvas.transform, true);
+
+        _potionSlots[index].SetPotion(newPotion);
+        newPotion.IncreaseCount();
+        _varietyAmount++;
+
+        newPotion.AddEvents(_disableEvent, _enableEvent, _decreaseEvent);
+    }
+
+    private bool CheckIfColorUsed(Color newColor)
+    {
+        foreach (PotionSlot slot in _potionSlots)
+        {
+            Potion potion = slot.GetPotion();
+
+            if(potion == null)
+            {
+                continue;
+            }
+
+            if (potion.Color == newColor)
+            {
+                potion.IncreaseCount();
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
I really love the structure of this code! I wouldn't change anything! Love the early exit using `continue` and `return true`.

You could use LINQ, technically, but I don't think this makes the code better:

```csharp
var slot = _potionSlots.Where(slot => slot.potion?.Color == newColor);
slot?.potion?.IncreaseCount();
return slot != null;
```